packages/contracts/test/diamond/facets/BondingCurveFacet.t.sol

```diff
@@ -125,28 +145,66 @@ contract ZeroStateBonding is BondingCurveFacetTest {
         assertEq(IBondingCurveFacet.poolBalance(), balance);
         assertEq(IDollar.balanceOf(IManager.treasuryAddress()), _amount);
     }
-}
 
-// contract DepositStateBonding is ZeroStateBonding {
-
-//     function setUp() public virtual override {
-//         super.setUp();
-
-//         uint256 collateralDeposited;
-//         uint32 connectorWeight = uint32(100);
-//         uint256 baseY = 100;
-
-//         assertEq(IBondingCurveFacet.poolBalance(), 0);
-//         vm.prank(admin);
-//         IBondingCurveFacet.setParams(
-//             connectorWeight,
-//             baseY
-//         ); 
-//         IBondingCurveFacet.deposit(
-//             collateralDeposited, 
-//             thirdAccount
-//         );
-//         // assertEq(IBondingCurveFacet.poolBalance(), collateralDeposited);
-
-//     }
-// }
+    function testPurchaseTargetAmountShouldRevertIfSupplyZero() public {
+        uint256 collateralDeposited;
+        uint connWeight;
+        uint256 poolBalance;
+        uint32 connectorWeight = uint32(bound(connWeight, 1, MAX_WEIGHT));
+
+        vm.expectRevert("ERR_INVALID_SUPPLY");
+        IBondingCurveFacet.purchaseTargetAmount(
+            collateralDeposited,
+            connectorWeight,
+            1,
+            0 
+        );
+    }
+
+    function testPurchaseTargetAmountShouldRevertIfParamsNotSet() public {
+        uint256 collateralDeposited;
+        uint256 bal;
+        uint256 poolBalance = bound(bal, 1, 1000000);
+        uint connWeight;
+        uint32 connectorWeight = uint32(bound(connWeight, 1, MAX_WEIGHT));
+
+        vm.expectRevert("ERR_INVALID_WEIGHT");
+        IBondingCurveFacet.purchaseTargetAmount(
+            collateralDeposited,
+            0,
+            1,
+            poolBalance 
+        );
+    }
+
+    function testPurchaseTargetAmount() public {
+        // Calculate expected result
+        uint256 tokensDeposited;
+        uint connWeight;
+        uint256 tokenIds;
+        uint32 connectorWeight = uint32(bound(connWeight, 1, MAX_WEIGHT));
+        uint256 bal;
```

Comment:
can provide this to input parameter so that this test is a fuzzing test

Reply:
Can you please clarify. I do not understand